istio/istio: `make docker.save` has warnings on MacOS

**Bug description**

Using MacOS, `make docker.save` has several warnings.

```
HUB=docker.io/ericvn \
		TAG=ericvn \
		PROXY_REPO_SHA=e4df956fb629490dbc7af43ec9c5edda3245d45f \
		VERSION=1.5-dev \
		DOCKER_ALL_VARIANTS="default distroless" \
		ISTIO_DOCKER_TAR=/Users/ericvn/work/go/src/istio.io/istio/out/linux_amd64/release/docker \
		BASE_VERSION=1.5-dev.1 \
		DOCKERX_PUSH= \
		./tools/buildx-gen.sh /Users/ericvn/work/go/src/istio.io/istio/out/linux_amd64/dockerx_build docker.pilot docker.proxytproxy docker.proxyv2 docker.app docker.app_sidecar docker.test_policybackend docker.mixer docker.mixer_codegen docker.citadel docker.galley docker.sidecar_injector docker.kubectl docker.node-agent-k8s docker.istioctl docker.operator
xargs: illegal option -- d
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
xargs: illegal option -- d
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
xargs: illegal option -- d
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
....
make: *** [dockerx.save] Error 1
```
The issue seems to be that ./tools/buildx-gen.sh has a few lines with an xarg -d command. The xarg command on MacOS does not have a -d parameter.



**Expected behavior**

`make docker.save` should complete without issues

**Steps to reproduce the bug**

Prepare MacOS development as specified.

Run `make docker.save`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

Current _master_ branch

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

MacOS. Catalina


Note that when the build moves to happening in a container, this problem will only happen when BUILD_WITH_CONTAINER=0.

--- a/tools/buildx-gen.sh
+++ b/tools/buildx-gen.sh
@@ -20,7 +20,7 @@ out="${1}"
 config="${out}/docker-bake.hcl"
 shift
 
-variants=$(for i in ${DOCKER_ALL_VARIANTS}; do echo "\"${i}\""; done | xargs -d'\n' | sed -e 's/ /, /g')
+variants=\"$(for i in ${DOCKER_ALL_VARIANTS}; do echo "\"${i}\""; done | xargs | sed -e 's/ /\", \"/g')\"
 cat <<EOF > "${config}"
 group "all" {
     targets = [${variants}]
@@ -30,7 +30,7 @@ EOF
 # Generate the top header. This defines a group to build all images for each variant
 for variant in ${DOCKER_ALL_VARIANTS}; do
   # Get all images. Transform from `docker.target` to `"target"` as a comma seperated list
-  images=$(for i in "$@"; do echo "\"${i#docker.}-${variant}\""; done | xargs -d'\n' | sed -e 's/ /, /g')
+  images=\"$(for i in "$@"; do echo "\"${i#docker.}-${variant}\""; done | xargs | sed -e 's/ /\", \"/g')\"
   cat <<EOF >> "${config}"
 group "${variant}" {
     targets = [${images}]

